listener/avalon_listener/tcs_listener.py

```diff
@@ -257,38 +287,56 @@ def parse_command_line(config, args):
         config['Logging']['LogFile'] = options.logfile
     if options.loglevel:
         config['Logging']['LogLevel'] = options.loglevel.upper()
-    if options.bind_uri:
-        bind_uri = options.bind_uri
+    if options.bind:
+        host_name, port = parse_bind_url(options.bind)
+    else:
+        if config.get("Listener") is None or \
+                config["Listener"].get("bind") is None:
+                    logger.warn("quit due to no suitable config for Listener")
+                    sys.exit(-1)
+        host_name, port = parse_bind_url(
+            config["Listener"].get("bind"))
     if options.lmdb_url:
-        config["KvStorage"]["remote_url"] = options.lmdb_url
+        config["KvStorage"]["remote_storage_url"] = options.lmdb_url
 
-    return bind_uri
+    return host_name, port
 
 # -----------------------------------------------------------------
 # -----------------------------------------------------------------
 
 
+def get_config_dir():
+    """
+    Returns the avalon configuration directory based on the
+    TCF_HOME environment variable (if set) or OS defaults.
+    """
+    if 'TCF_HOME' in os.environ:
+        return os.path.join(os.environ['TCF_HOME'], 'listener/')
+
+    return '/etc/avalon'
```

Comment:
If TCF_HOME is not present log error and exist. Let's avoid hard coded path like '/etc/avalon'.

Reply:
Let's make debian related changes in a separate PR. We need to address this in other modules too. For now, let's log and exist if TCF_HOME directory is not set

Reply:
I am following the approach followed in https://github.com/hyperledger/avalon/blob/master/examples/shared_kv_storage/kv_storage/remote_lmdb/lmdb_listener.py#L127

Reply:
Hard coding the path doesn't look good to me. We can raise a separate PR to remove the changes in lmdb listener

Reply:
Debians will eventually come for Avalon. If somebody has already put some effort to handle that in code, we should not remove that. and for debians we need a path where we expect config files to be copied to. So, we need a path.

Reply:
Hard coding a path in code is a not a recommended coding practice. There are other approaches available to keep configuration files in /etc folder for debian if keeping at that path is absolutely must. For example, create a symlink in /etc pointing to $TCF_HOME. Another approach is keep a module specific config path and use that in code.

Reply:
@manojgop There can be many other ways too. As per community feedback (on various occasions), they like sawtooth because of the smooth and clear deployment process using debians. In that, there is just one path '/etc/sawtooth' which they have to go and update the config files according to their setup. Your suggestions will ask users to do some extra stuff (creating symlinks etc). And if they are using different VMs for different services, it will be even more work for them. Nobody likes hard coding but a standard way which makes life easy for everyone makes sense and we have experienced that in the past.

Reply:
symlink can be created via installation script not manually. BTW it is not mandatory to keep the config files in /etc/avalon even for debians. We can still decide to keep under $TCF_HOME. That way we use a similar logic for config file for both standalone and docker based builds. For example, checkout https://www.debian.org/doc/manuals/maint-guide/dother.en.html

Reply:
@pankajgoyal2 We don't have any debians for Avalon for now. we can makes debian specific changes in separate PR. for now it's much easy to assume that TCF_HOME path is set via config file and return an error if path is not set.